Source/Model/Classes/NMR_Model.cpp

```diff
@@ -391,19 +396,24 @@ namespace NMR {
 		}
 	}
 
+    ModelResourceID CModel::generateResourceID()
+    {
+        // Determine the initial resource ID
+        ModelResourceID currentResourceID = 1;
+        auto iIterator = m_ResourceMap.rbegin();
+        if (iIterator != m_ResourceMap.rend()) {
+            currentResourceID = iIterator->first + 1;
+        }
 
-	// Retrieve a unique Resource ID
-	ModelResourceID CModel::generateResourceID()
-	{
-		// TODO: is this truly safe?
-		auto iIterator = m_ResourceMap.rbegin();
-		if (iIterator != m_ResourceMap.rend())
-			return iIterator->first + 1;
-		else
-			return 1;
-	}
+        // Ensure the generated ID is unique
```

Comment:
Do we need to update m_maxResourceId here, if currentResourceId > m_maxResourceId ?